Switch to actually calling stage('foo') for parallel stages

abayer opened the pull request:
* JENKINS issue(s):
    * n/a
* Description:
    * I discovered that this won't actually mess with Blue Ocean visualization, since Blue Ocean already discards stages inside parallel branches but keeps the steps inside said stages. So...let's simplify things.
* Documentation changes:
    * n/a
* Users/aliases to notify:
    * @reviewbybees

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

abayer commented:
@kzantow @vivek - I cc'd you on this to make sure that I'm not glaringly wrong from the Blue Ocean side. I checked `PipelineNodeTest` there, which has a nested stages test, and this behavior should match what's expected there.

rsandell commented on pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy:
```diff
@@ -150,66 +146,68 @@ public class ModelInterpreter implements Serializable {
-                        }
+            script.stage(thisStage.name) {
+                try {
+                    // NOTE - this will switch to script.stage in the future.
```
The future is now!

abayer commented:
@vivek I'll do a SNAPSHOT deploy and open a PR against Blue Ocean using that SNAPSHOT to go through the tests/ATH there.

abayer commented:
Opened https://github.com/jenkinsci/blueocean-plugin/pull/1308 using the deployed SNAPSHOT.

abayer commented:
One legit failure in Blue Ocean so far - we're not marking the parallel branch above the nested stage as skipped etc, so that needs to be added. There were also some step counts that needed to be fixed now that the `properties` step isn't always executed.

vivek commented:
Sounds good, thanks @abayer

abayer commented:
@vivek Ok, think I got it - I tweaked things to propagate the `StageStatus` tags from the nested `stage` inside the `parallel` branch to the `parallel` branch too, and added logic so that if a `stage` containing nested `stage`s is skipped due to failure/unstable/conditional, we'll still "run" its child stages but mark them as skipped for the same reason (just as we would with a non-nested stage). I updated the Blue Ocean PR too.

abayer commented:
@vivek Since Blue Ocean looks good with that PR, is this good to go from your perspective? If so, I'll merge this, update #174 on top of master, do a new SNAPSHOT deploy, and update the Blue Ocean PR to use that SNAPSHOT so that it's even closer to what'll actually be Declarative 1.2.
